aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/entities/TraceID.java

```diff
@@ -95,6 +65,46 @@ public static TraceID fromString(String xrayTraceId) {
         return result;
     }
 
+    /**
+     * Returns an invalid {@link TraceID} which can be used when an ID is needed outside the context of a trace, for example for
+     * an unsampled segment.
+     */
+    public static TraceID invalid() {
+        return INVALID;
+    }
+
+    private static final int TRACE_ID_LENGTH = 35;
+    private static final int TRACE_ID_DELIMITER_INDEX_1 = 1;
+    private static final int TRACE_ID_DELIMITER_INDEX_2 = 10;
+
+    private static final char VERSION = '1';
+    private static final char DELIMITER = '-';
+
+    private BigInteger number;
+    private long startTime;
```

Comment:
To clarify, I mostly disabled the check to avoid the static methods being below the private constructors :)

I think it's a tradeoff of readability of the API vs the code itself - skimming from the top to quickly find the entrypoints of the class is quite nice. And even GitHub now adays makes it easy to jump to definitions that may be below so I'd say the negative aspect is not so significant.

Reply:
Ok - mostly an opinion thing so it's fine as is.